lms/djangoapps/support/static/support/jsx/entitlements/data/api/client.js

```diff
@@ -0,0 +1,55 @@
+import 'whatwg-fetch'; // fetch polyfill
+import Cookies from 'js-cookie';
+
+import endpoints from './endpoints';
+
+export function requestEntitlements(username_or_email) {
+	console.log('called requestEntitlements api function');
+  return fetch(
+    `${endpoints.entitlementList}/${username_or_email}`, {
+      credentials: 'same-origin',
+      method: 'get'
+    },
+  );
+}
+
+export function createEntitlement(course_uuid, user, mode, reason, comments) {
```

Comment:
I've been finding that using objects as method parameters makes things, in some cases, easier to reason / read about when that method is called.

[Here's an article that provides some examples](https://medium.freecodecamp.org/elegant-patterns-in-modern-javascript-roro-be01e7669cbd).

```javascript
// might be more readable
createEntitlement({ courseUUID: '098sfgoj5', user: 'baejadley', mode: 'baemode', reason: 'tobebae', comments: 'jaebaebae' });
// than
createEntitlement('098sfgoj5', 'baejadley', 'baemode', 'tobebae', 'jaebaebae' );

```

Reply:
I like it